examples/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect/README.md

```diff
@@ -0,0 +1,289 @@
+
+- Have the latest [Arduino IDE](https://www.arduino.cc/en/Main/Software) installed.
+
+- Have the [Arduino CLI installed](https://arduino.github.io/arduino-cli/0.21/installation/). 
+
+- Have one of the following interfaces to your Azure IoT Hub set up:
+  - [Azure Command Line Interface](https://docs.microsoft.com/cli/azure/install-azure-cli?view=azure-cli-latest) (Azure CLI) utility installed, along with the [Azure IoT CLI extension](https://github.com/Azure/azure-iot-cli-extension).
+
+    On Windows:
+
+      Download and install: https://aka.ms/installazurecliwindows
+
+      ```powershell
+      PS C:\>az extension add --name azure-iot
+      ```
+
+    On Linux:
+
+      ```bash
+      $ curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+      $ az extension add --name azure-iot
+      ```
+
+      A list of all the Azure IoT CLI extension commands can be found [here](https://docs.microsoft.com/cli/azure/iot?view=azure-cli-latest).
+
+  - **Optional**: The most recent version of [Azure IoT Explorer](https://github.com/Azure/azure-iot-explorer/releases) installed. More instruction on its usage can be found [here](https://docs.microsoft.com/azure/iot-pnp/howto-use-iot-explorer).
+
+  NOTE: This guide demonstrates use of the Azure CLI and does NOT demonstrate use of Azure IoT Explorer.
+
+## IoT Hub Setup
+1. In the Azure portal, navigate to your IoT Hub resource.
+
+1. On the left side menu under **Device management**, click on **'Devices'**.
+
+1. Select **'+ Add Device'**.
+
+1. Give your device a unique name.
+
+1. Select 'Symmetric key', 'Auto-Generate keys', and 'Enable' for Connecting the device to an IoT Hub, then click 'Save'.
+
+1. When the device has been created, select its name under 'Device ID'.
+
+1. We'll need the following information from the device page:
+
+    - Device ID
+    - Primary Key
+    - HostName (under Primary Connection String)
+
+_NOTE: Device keys are used to automatically generate a SAS token for authentication, which is only valid for one hour._
+
+## Arduino IDE Setup
+
+1. Open the Arduino IDE.
+
+1. Install Arduino Mbed OS Nano Boards support in the Arduino IDE. [Full instructions can be found here.](https://docs.arduino.cc/hardware/nano-rp2040-connect)
+
+    - Navigate to **Tools > Board > Board Manager**
+    - Search for 'RP2040' and install the **Arduino Mbed OS Nano Boards** core.
+    - Always install the latest version.    
+    
+    *Note: This process may take several minutes.*  
+
+1. Nagivate to **Tools > Board > Arduino Mbed OS Nano Boards** and select **'Arduino Nano RP2040 Connect'**.
+
+1. If this is your first time using the Nano RP2040 Connect, [follow these instructions to update the WiFi firmware on the Nano RP2040 Connect](https://docs.arduino.cc/tutorials/nano-rp2040-connect/rp2040-upgrading-nina-firmware).
+
+1. Install WiFiNINA library for the Nano RP2040 Embedded C SDK sample. 
```

Comment:
I've learned this step (adding the wifiNina lib is actually a pre-requisite to the firmware update, so this step should be before that. 
Also, for completeness, we need to add the need for two additional libraries: ArduinoBearSSL and ArduinoMqttClient which are needed but not mentioned in the guide.